src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -1435,48 +1454,37 @@ private RepairSchedule createRepairScheduleFromRow(Row repairScheduleRow) {
 
   @Override
   public boolean updateRepairSchedule(RepairSchedule newRepairSchedule) {
-    final Set<UUID> repairHistory = Sets.newHashSet();
-    repairHistory.addAll(newRepairSchedule.getRunHistory());
     RepairUnit repairUnit = getRepairUnit(newRepairSchedule.getRepairUnitId());
-    List<ResultSetFuture> futures = Lists.newArrayList();
-
-    futures.add(
-        session.executeAsync(
-            insertRepairSchedulePrepStmt.bind(
-                newRepairSchedule.getId(),
-                newRepairSchedule.getRepairUnitId(),
-                newRepairSchedule.getState().toString(),
-                newRepairSchedule.getDaysBetween(),
-                newRepairSchedule.getNextActivation(),
-                repairHistory,
-                newRepairSchedule.getSegmentCount(),
-                newRepairSchedule.getRepairParallelism().toString(),
-                newRepairSchedule.getIntensity(),
-                newRepairSchedule.getCreationTime(),
-                newRepairSchedule.getOwner(),
-                newRepairSchedule.getPauseTime(),
-                newRepairSchedule.getSegmentCountPerNode())));
-
-    futures.add(
-        session.executeAsync(
-            insertRepairScheduleByClusterAndKsPrepStmt.bind(
-                repairUnit.getClusterName(), repairUnit.getKeyspaceName(), newRepairSchedule.getId())));
-
-    futures.add(
-        session.executeAsync(
-            insertRepairScheduleByClusterAndKsPrepStmt.bind(
-                repairUnit.getClusterName(), " ", newRepairSchedule.getId())));
-
-    futures.add(
-        session.executeAsync(
-            insertRepairScheduleByClusterAndKsPrepStmt.bind(
-                " ", repairUnit.getKeyspaceName(), newRepairSchedule.getId())));
-
-    try {
-      Futures.allAsList(futures).get();
-    } catch (InterruptedException | ExecutionException ex) {
-      LOG.error("failed to quorum update repair schedule " + newRepairSchedule.getId(), ex);
-    }
+    BatchStatement batchStatement = new BatchStatement();
```

Comment:
why do we need to change this to batch in this PR?

Reply:
I ran into some issues when testing this part of the code. The async queries were returning errors that I couldn't figure out. I switched to a batch statement and it worked fine 🤷  I tried to debug this for a while without managing to understand the root cause, which is very unsatisfying 😕 
Nonetheless, I think we're better with a batch statement here for consistency reasons and keeping in mind that schedule creation shouldn't be happening very often.

Reply:
i'm uneasy about putting batch statements into people's clusters, like I am about LWTs. Reaper ideally should aim to be as light as possible, as folk do use their production C* as the Reaper backend… 

Can we resume debugging…? (sorry to be a pain)
A series of strongly consistent requests should not be causing a problem, and we would absolutely benefit from knowing why it is not working in practice…

Reply:
Whatever happened when I was testing this is now gone and I can't reproduce the exception. We're back to the previous version of this method using async queries.